src/EditorFeatures/Core/Implementation/Classification/CopyPasteAndPrintingClassificationBufferTaggerProvider.Tagger.cs

```diff
@@ -104,10 +108,12 @@ private TagSpanIntervalTree<IClassificationTag> CachedTags
 
             private void OnEventSourceChanged(object sender, TaggerEventArgs _)
             {
-                _owner._notificationService.RegisterNotification(
-                    OnEventSourceChanged_OnForeground,
-                    _owner._asyncListener.BeginAsyncOperation("SemanticClassificationBufferTaggerProvider"),
-                    _cancellationTokenSource.Token);
+                _owner.ThreadingContext.JoinableTaskFactory.RunAsync(async () =>
+                {
+                    using var _ = _owner._asyncListener.BeginAsyncOperation("SemanticClassificationBufferTaggerProvider");
+                    await _owner.ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(_cancellationTokenSource.Token);
```

Comment:
❓ Is there any value of calling to the main thread for the _sole_ purpose of clearing `CachedTags`?

Reply:
Currently, it ensures that state management here has no race conditions.  All reads/writes to these values are on the UI thread.

I think we could certainly clear on the BG, but we have to be more careful that then the reads on the UI thread are consistent.  This is part of the work i planned for later on, but dont' want to touch just yet because of risk of regression.  For CopyPAsteTagger it's likely very safe to go in and tweak.  However, for the general tagger case I want to be very careful as races in taggers are *awful* to discover :)

Reply:
@CyrusNajmabadi I see in the end we went with https://github.com/dotnet/roslyn/pull/52825 which just switched over to a lock?

Reply:
Yup.  I didn't want to have both in teh same PR.  so i split into separate.